deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java

```diff
@@ -148,4 +145,44 @@ public void vpTreeTest() {
 
     }
 
+    @Test
+    public void testVPSearchOverNaturals() throws Exception {
+        final int firstPoint = 0;
+        final int lastPoint = 20;
+        final int queryPoint = 12;
+        final int k = 5;
+
+        INDArray points = Nd4j.arange(firstPoint, lastPoint).transpose();
+        INDArray query = Nd4j.zeros(1);
+        query.putScalar(0, queryPoint);
+
+        INDArray trueResults = Nd4j.zeros(k);
+        for (int i = 0; i < k; i++) {
+            int pt = queryPoint - k / 2 + i;
+            trueResults.putScalar(i, pt);
+        }
+        VPTree tree = new VPTree(points);
+        List<DataPoint> results = new ArrayList<>();
+        List<Double> distances = new ArrayList<>();
+        tree.search(query, k, results, distances);
+        INDArray sortedResults = Nd4j.zeros(k);
+        int i = 0;
+        for (DataPoint p : results)
+            sortedResults.put(new int[] {0, i++}, p.getPoint());
+        Nd4j.sort(sortedResults, 1, true);
+        if (!trueResults.equalsWithEps(sortedResults, 1E-12))
```

Comment:
You should use junit's `assertTrue` and assertEquals`, see:
http://junit.org/junit4/javadoc/4.8/org/junit/Assert.html

They take a String argument for displaying a message, in which you can print the received values (as opposed to expected)